mutex_flock.go

```diff
@@ -67,8 +68,45 @@ func acquireFlock(name string, done <-chan struct{}) <-chan acquireResult {
 	}
 
 	flockName := filepath.Join(os.TempDir(), "juju-"+name)
+	chownFromRoot := func() (err error) {
+		if cmd, ok := os.LookupEnv("SUDO_COMMAND"); ok && cmd != "" {
+			var uid, gid int
+			uid, err = strconv.Atoi(os.Getenv("SUDO_UID"))
+			if err != nil {
+				err = errors.Annotate(err, "parsing SUDO_UID")
```

Comment:
Even with named return values, I prefer the explicit approach.

Reply:
Fine by me, I'll get on it!